ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/internal/EditorConfigGenerator.kt

```diff
@@ -0,0 +1,66 @@
+package com.pinterest.ktlint.core.internal
+
+import com.pinterest.ktlint.core.EditorConfig
+import com.pinterest.ktlint.core.RuleSet
+import com.pinterest.ktlint.core.api.FeatureInAlphaState
+import com.pinterest.ktlint.core.api.UsesEditorConfigProperties
+import com.pinterest.ktlint.core.api.UsesEditorConfigProperties.Companion.getEditorconfigPropertyValue
+import java.nio.file.Path
+
+/**
+ * Generates Kotlin section content for `.editorconfig` file.
+ *
+ * Rule should implement [UsesEditorConfigProperties] interface to support this.
+ */
+@FeatureInAlphaState
+internal class EditorConfigGenerator(
+    private val editorConfigLoader: EditorConfigLoader
+) {
+    /**
+     * Method loads merged `.editorconfig` content from [Params.fileName] path,
+     * and then, by querying rules from [Params.ruleSets] for missing properties default values,
+     * generates Kotlin section (default is `[*.{kt,kts}]`) new content.
+     *
+     * @return Kotlin section editorconfig content. For example:
+     * ```properties
+     * final-newline=true
+     * indent-size=4
+     * ```
+     */
+    fun generateEditorconfig(
+        filePath: Path,
+        ruleSets: Iterable<RuleSet>,
+        debug: Boolean = false
+    ): String {
+        val editorConfig = EditorConfig.fromMap(
+            editorConfigLoader.loadPropertiesForFile(
+                filePath = filePath,
+                debug = debug
+            )
+        )
+
+        return ruleSets
+            .flatMap {
+                it.rules.toList()
+            }
+            .fold(mutableMapOf<String, String?>()) { acc, rule ->
+                if (rule is UsesEditorConfigProperties) {
+                    if (debug) System.out.println("Checking properties for '${rule.id}' rule")
+                    rule.editorconfigProperties.forEach { prop ->
+                        if (debug) System.out.println("Setting '$prop' property value")
+                        acc[prop] = rule.getEditorconfigPropertyValue(
+                            editorConfig,
+                            prop
+                        )
+                    }
+                }
+
+                acc
+            }
+            .filterValues { it != null }
+            .map {
+                "${it.key} = ${it.value}"
+            }
+            .joinToString(separator = "\n")
```

Comment:
I have a feeling that on windows it should be something like `\r\n`, but not sure

Reply:
Fixed via a3fbfe4  :+1: